Client/States/GameplayState.cpp

```diff
@@ -99,13 +104,66 @@ void GameplayState::Update(float dt) {
 void GameplayState::ReadNetworkFunctions() {
     while (!networkData->incomingFunctions.IsEmpty()) {
         FunctionPacket packet = networkData->incomingFunctions.Pop();
-        if (packet.functionId == Replicated::AssignPlayer) {
-            DataHandler handler(&packet.data);
-            auto networkId = handler.Unpack<int>();
-            AssignPlayer(networkId);
+        DataHandler handler(&packet.data);
+        switch (packet.functionId) {
+            case(Replicated::AssignPlayer): {
+                int networkId = handler.Unpack<int>();
+                AssignPlayer(networkId);
+            }
+            break;
+
+            case(Replicated::Camera_GroundedMove): {
+                float intesnity = handler.Unpack<float>();
+                playerMovement = intesnity;
+            }
+            break;
+
+            case(Replicated::Camera_Jump): {
+                jumpTimer = 3.14f;
+            }
+            break;
+            
+            case(Replicated::Camera_Land): {
+                float grounded = handler.Unpack<float>();
+                landIntensity = std::clamp(grounded, 0.0f, landFallMax);
+                landTimer = 3.14f;
+            }
+            break;
+            
+            case(Replicated::Camera_Strafe): {
+                float strfSpd = handler.Unpack<float>();
+                strafeSpeed = strfSpd;
+            }
+            break;
         }
+
     }
 }
+void GameplayState::ResetCameraAnimation() {
+    playerMovement = playerMovement * 0.95f;
+    strafeSpeed = 0.0f;
+}
+void GameplayState::WalkCamera(float dt) {
+    world->GetMainCamera()->SetOffsetPosition(Vector3(0, abs(-0.015f + 0.1f*sin(walkTimer)) * (playerMovement/15.0f), 0));
+    walkTimer += dt * playerMovement * 0.75f;
+}
+
+void GameplayState::JumpCamera(float dt) {
+    world->GetMainCamera()->SetOffsetPosition(world->GetMainCamera()->GetOffsetPosition() + Vector3(0, -0.55f * sin(3.14f - jumpTimer), 0));
+    jumpTimer = std::clamp(jumpTimer - dt * 18.0f, 0.0f, 3.14f);
+}
+
+void GameplayState::LandCamera(float dt) {
+    world->GetMainCamera()->SetOffsetPosition(world->GetMainCamera()->GetOffsetPosition() + 
+                                              Vector3(0, -0.2f * sin(3.14f - 3.14f * sin(3.14f/2 - landTimer/2)) / landFallMax * landIntensity, 0));
+    landTimer = std::clamp(landTimer - dt * 15.0f, 0.0f, 3.14f);
+}
+
+void GameplayState::StrafeCamera(float dt) {
+    strafeAmount = strafeAmount * 0.95f + strafeSpeed * 0.05f;
+    //std::cout << strafeAmount << "\n";
+    world->GetMainCamera()->SetRoll(-strafeTiltAmount * (strafeAmount / strafeSpeedMax));
+}
```

Comment:
Lots of magic numbers here, it would be useful to #definte these to make their purpose clearer.